### PR TITLE
Add sort DG

### PR DESCRIPTION
**Pull request by naranghardik16**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -307,7 +307,33 @@ The following activity diagram summarizes what happens when the user inputs a fi
...
+
+The `SortCommandParser` class implements the `Parser<SortCommand>` interface. It manages the parsing of the arguments (sorting type and sorting order in the case of a sort command) in the user input. The `parse` method in `SortCommandParser` first checks if the sorting order has been given or not. It uses a BiFunction function that accepts two Long variables and produces an Integer result. This function decides the sorting order. If the sorting order is not specified, then ascending order is chosen to be the default option. If an invalid sorting order is entered by the user, a `ParseException` is thrown with the feedback message of Invalid Command Format and correct usage message for the sort command. The second if condition checks if the sorting type has been given or not. If the sorting type has not been specified, then time is chosen to be the default option. If an invalid sorting type is entered by the user, a `ParseException` is thrown with the feedback message of Invalid Command Format and correct usage message for the sort command. It then returns a `SortCommand` back to `UniFyParser`, applying the sort function to the given arguments.
+
+##### Usage Scenario
```
Comment (KelvinSoo) [inline docs/DeveloperGuide.md:326]: It would be nice if this follows our current DG format.

Comment (naranghardik16) [reply docs/DeveloperGuide.md:326]: Okay sure I'll update it.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -307,7 +307,33 @@ The following activity diagram summarizes what happens when the user inputs a fi
...
+
+The following demonstrates a usage scenario where the user wants to delete the first, second and third item in her/his task list.
+
+1. The method `execute("sort x/time o/asc")` inside LogicManager calls the `parseCommand`method of `UniFyParser`.
+2. `parseCommand` in `UniFyParser` takes in the String “sort x/time o/asc” as its parameter and initializes a `SortCommandParser` object.
+3. It then calls the `parse` method in `SortCommandParser` to parse the string `”x/time o/asc”`.
```
Comment (lwj1711) [inline docs/DeveloperGuide.md:332]: I think the first quotation mark should be the open one

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -307,7 +307,33 @@ The following activity diagram summarizes what happens when the user inputs a fi
...
+
+##### Usage Scenario
+
+The following demonstrates a usage scenario where the user wants to delete the first, second and third item in her/his task list.
+
+1. The method `execute("sort x/time o/asc")` inside LogicManager calls the `parseCommand`method of `UniFyParser`.
```
Comment (lwj1711) [inline docs/DeveloperGuide.md:330]: Might be lacking a space after `parseCommand`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -307,7 +307,33 @@ The following activity diagram summarizes what happens when the user inputs a fi
...
+2. `parseCommand` in `UniFyParser` takes in the String “sort x/time o/asc” as its parameter and initializes a `SortCommandParser` object.
+3. It then calls the `parse` method in `SortCommandParser` to parse the string `”x/time o/asc”`.
+4. A `SortCommand` object will be initialized, taking in the sort type and sort order in the function, in this case the type being time and order being asc (ascending order).
+5. The method call then returns to `LogicManager`, which calls the `execute` method of `SortCommand`.
+6. By using a functional interface called ‘BiFunction’, the `SotCommand` sorts the task list by calling a method ‘sortTasks’ under ‘Model’.
```
Comment (lwj1711) [inline docs/DeveloperGuide.md:335]: "SotCommand" --> "SortCommand"

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (KelvinSoo) [reply docs/DeveloperGuide.md:326]: I will change the format of find

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (naranghardik16) [reply docs/DeveloperGuide.md:335]: okay will change that. Thanks for pointing out.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (naranghardik16) [reply docs/DeveloperGuide.md:330]: okay will add a space.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (naranghardik16) [reply docs/DeveloperGuide.md:332]: Yes I'll change that. Thanks for pointing out.

**Pull request merged**